RPT_Application/src/main/java/com/harman/user_database/Employee.java

```diff
@@ -0,0 +1,157 @@
+
+@Entity
+@Table(name = "employee")
+public class Employee implements Cloneable {
+
+	@Id
+	private String id;
+
+	@NotEmpty
+	@Column(unique = true)
+	private String username;
+
+	@NotEmpty
+	private String password;
+
+	private String authority;
+
+	@ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+	@JoinTable(name = "employee_authority", joinColumns = { @JoinColumn(name = "employee_id") }, inverseJoinColumns = {
+			@JoinColumn(name = "authority_id") })
+	public Set<Authority> authorities = new HashSet<>();
+
+	@OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL, orphanRemoval = true)
+	private List<Report> reports = new ArrayList<>();
+
+	public Employee() {
+		super();
+	}
+
+	public Employee(String username, String password) {
+		super();
+		this.username = username;
+		this.password = password;
+	}
+
+	public Employee(@NotEmpty String username, @NotEmpty String password, Authority authority) {
+		super();
+		this.username = username;
+		this.password = password;
+		this.authorities.add(authority);
+
+		if (authority.getName().equals(AuthorityType.ROLE_USER)) {
+			this.authority = AuthorityType.ROLE_USER.toString();
+		} else {
+			this.authority = AuthorityType.ROLE_ADMIN.toString();
+		}
+	}
+//
+//	public Employee(@NotEmpty String username, @NotEmpty String password, Authority authority, List<Report> reports) {
+//		super();
+//		this.username = username;
+//		this.password = password;
+//		this.authorities.add(authority);
+//		this.reports = reports;
+//
+//		if (authority.getName().equals(AuthorityType.ROLE_USER)) {
+//			this.authority = AuthorityType.ROLE_USER.toString();
+//		} else {
+//			this.authority = AuthorityType.ROLE_ADMIN.toString();
+//		}
+//	}
+
+	public Employee(String id, @NotEmpty String username, @NotEmpty String password, Authority authority) {
+		super();
+		this.id = id;
+		this.username = username;
+		this.password = password;
+		this.authorities.add(authority);
+
+		if (authority.getName().equals(AuthorityType.ROLE_USER)) {
+			this.authority = AuthorityType.ROLE_USER.toString();
+		} else {
+			this.authority = AuthorityType.ROLE_ADMIN.toString();
+		}
+	}
+
+	@Override
+	public Employee clone() throws CloneNotSupportedException {
+		return (Employee) super.clone();
+	}
+
+	public String getId() {
+		return id;
+	}
+
+	public void setId(String id) {
+		this.id = id;
+	}
+
+	public String getUsername() {
+		return username;
+	}
+
+	public void setUsername(String username) {
+		this.username = username;
+	}
+
+	public String getPassword() {
+		return password;
+	}
+
+	public void setPassword(String password) {
+		this.password = password;
+	}
+
+	public String getAuthority() {
+		return authority;
+	}
+
+	public void setAuthority(AuthorityType auth) {
+		this.authority = auth.name();
+	}
+
+	public Collection<? extends GrantedAuthority> getAuthorities() {
+		return (Collection<? extends GrantedAuthority>) this.authorities;
+	}
+
+	public String getReports() {
```

Comment:
This should be rather getReportsAsString(). GetReports() suggests that we will have a collection of reports.

Reply:
yes, of course